miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/FileWriter.kt

```diff
@@ -1,22 +1,45 @@
 package com.rakuten.tech.mobile.miniapp.storage
 
+import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
+import java.io.BufferedOutputStream
 import java.io.File
+import java.io.FileOutputStream
 import java.io.InputStream
+import java.util.zip.ZipEntry
+import java.util.zip.ZipInputStream
 
-internal class FileWriter {
+internal class FileWriter(private val coroutineDispatcher: CoroutineDispatcher = Dispatchers.IO) {
 
-    suspend fun write(inputStream: InputStream, path: String) = withContext(Dispatchers.IO) {
-        inputStream.toFile(path)
+    suspend fun unzip(inputStream: InputStream, zipPath: String) = withContext(coroutineDispatcher) {
+        ZipInputStream(inputStream).decompress(zipPath)
     }
 }
 
-internal fun InputStream.toFile(path: String) {
-    use { input ->
-        File(path).apply {
-            parentFile?.mkdirs()
-            outputStream().use { input.copyTo(it) }
+internal fun ZipInputStream.decompress(zipPath: String) = use { input ->
+    val outputFilePath = File(zipPath).apply { parentFile?.mkdirs() }
+    var entry: ZipEntry?
+
+    while (input.nextEntry.also { entry = it } != null) {
+        val newFile = File(outputFilePath.parentFile, entry!!.name)
+        if (entry!!.isDirectory)
+            newFile.mkdir()
```

Comment:
I know that there is `SecurityException` in document but what is the real case that `SecurityException` can be thrown 🤔 ?

Reply:
`SecurityException` is a rare case unless System set any security policy with given file's properties. In this case, as the document mentioned that it will show the exception while `checkWrite(...)` will be invoked by SecurityManager. It's not obvious at-least right now if there is no spec about setting security policy.

Reply:
The place which we stored miniapp is writeable so we don't need to catch on `SecurityException`.